Modules/posixmodule.c

```diff
@@ -5088,384 +5088,407 @@ os__path_splitroot_impl(PyObject *module, path_t *path)
 }
 
 
-/*[clinic input]
-os._path_isdir
-
-    s: 'O'
+#define PY_IFREG  1 // Regular file
+#define PY_IFDIR  2 // Directory
+#define PY_IFLNK  4 // Symlink
+#define PY_IFMNT  8 // Mount Point (junction)
+#define PY_IFLRP 16 // Link Reparse Point (name-surrogate, symlink, junction)
```

Comment:
`PY_IFLRP` and related code could be removed if you want. There's no high-level `os.path` test for name-surrogate reparse points in general -- just for symlinks and junctions. At a low level, `follow_symlinks` is based on name surrogates, so it's possible to check for them via the `st_reparse_tag` value from `os.lstat()`.

Reply:
I guess we could leave it in, if we ever want to provide `os.path.isanylink()` (an alias for `os.path.islink()` on POSIX) in case you don't care what type of link it is. This could be discussed after this is merged.